### PR TITLE
Remove utf-8 string decoding from normalizers

### PR DESCRIPTION
**Pull request by dylanjw**
### What was wrong?
The eth-abi v2 alpha release adds utf-8 string decoding to the string type decoder.  String decoding should therefore be removed from web3 return data normalizers.

#### Cute Animal Picture

image

### CODE REVIEW

**Comment by dylanjw** [comment]
@pipermerriam Updated to add support for eth-abi >=2 without removing support for <2.

File: web3/utils/normalizers.py [hunk 1]
```diff
@@ -140,11 +144,14 @@ def abi_ens_resolver(w3, abi_type, val):
 
 
 BASE_RETURN_NORMALIZERS = [
-    addresses_checksummed,
-    decode_abi_strings,
+    addresses_checksummed
```
Comment (carver) [inline web3/utils/normalizers.py]: *style nitpick* leave the trailing comma.

File: tests/core/contracts/test_contract_call_interface.py [hunk 2]
```diff
@@ -184,6 +188,9 @@ def test_call_get_string_value(string_contract, call):
     assert result == "Caqalai"
 
 
+@pytest.mark.skipif(
+    LooseVersion(eth_abi.__version__) >= LooseVersion("2"),
+    reason="eth-abi >=2 does utf-8 string decoding")
 def test_call_read_string_variable(string_contract, call):
     result = call(contract=string_contract,
                   contract_function='constValue')
```
Comment (carver) [inline tests/core/contracts/test_contract_call_interface.py]: Maybe add a second test that shows what happens on an ABI decoding error. We should probably do something like capture the eth-abi error and re-raise a generic `ValueError`.

Comment (dylanjw) [reply tests/core/contracts/test_contract_call_interface.py]: It raises a 'BadFunctionCallOutput' error

**Comment by dylanjw** [comment]
@pipermerriam @carver Ready for another review

File: setup.py [hunk 3]
```diff
@@ -19,7 +19,7 @@
     install_requires=[
         "toolz>=0.9.0,<1.0.0;implementation_name=='pypy'",
         "cytoolz>=0.9.0,<1.0.0;implementation_name=='cpython'",
-        "eth-abi>=1.1.1,<2",
+        "eth-abi==2.0.0a1",
```
Comment (carver) [inline setup.py]: We shouldn't require people to upgrade. We especially shouldn't require people to install an alpha dependency in a stable release. `eth-abi>=1.1.1,<3` seems like a reasonable choice

**Comment by dylanjw** [comment]
@pipermerriam @carver Is this good to merge?

**Pull request merged**